stablehlo/dialect/Serialization.cpp

```diff
@@ -53,17 +56,13 @@ LogicalResult serializePortableArtifact(ModuleOp module,
   }
 
   // Write bytecode with producer string "StableHLO_vX.Y.Z"
+  // using the bytecode format version associated with the StableHLO release.
   auto producer = "StableHLO_v" + targetVersion.str();
   BytecodeWriterConfig writerConfig(producer);
-  // bytecodeVersion = 1 is what has been predominantly used in practice to
-  // serialize portable StableHLO artifacts.
-  // Theoretically speaking, StableHLO v0.9.0 which introduced compatibility
-  // guarantees was released on 3/2/2023 and bytecodeVersion = 1 was released
-  // on 3/10/2023, so there was a time period when we guaranteed compatibility
-  // for StableHLO consumers which only supported bytecodeVersion = 0.
-  // However, this time period (1 month of forward compatibility) has expired,
-  // so it's fine to hardcode bytecodeVersion = 1 here.
-  writerConfig.setDesiredBytecodeVersion(1);
+  auto bytecodeVersion = vhlo::Version::getBytecodeFormatVersion(
+      vhlo::Version::fromString(targetVersion).value());
```

Comment:
What happens if `vhlo::Version::fromString(targetVersion)` fails?

Reply:
Would have errored out a few lines above in:

```
pm.addPass(stablehlo::createVhloToVersionPass({targetVersion.str()}));
```

Reply:
Ah, you're right. I think it's fine to not do an extra round of validation here then.